src/main/java/com/hivemq/cli/mqtt/ContextClientDisconnectListener.java

```diff
@@ -48,5 +54,12 @@ else if (ShellCommand.DEBUG) {
             }
         }
 
+        MqttClientExecutor.getClientDataMap().remove(getKeyFromConfig(context.getClientConfig()));
```

Comment:
Why is the client data removed. I thought, disconnected clients remain in the map?

Reply:
I decided on removing clients from the map when they get disconnected because I think that keeping the disconnected clients will only cause headaches in the long run with a very low benefit. 🤔